### PR TITLE
Wire up all the contract data to the funding page

### PR DESCRIPTION
**Pull request by vipergtsrz**
Resolved all the "TODO"s in the funding page and hooked everything up to the DealTokenSwap methods

### CODE REVIEW

File: src/funding/funding.ts [hunk 1]
```diff
@@ -186,32 +199,37 @@ export class Funding {
...
-    //TODO handle account provider transaction rejection
-    //TODO handle the popup notification on the event of the deposit actually being completed
+    //all validation passed so submit the deposit
+    const depositTransaction = await this.deal.depositTokens(this.firstDao, depositToken.address, this.depositAmount);
+    if (depositTransaction){
+      //deposit was successful
+      this.eventAggregator.publish("handleInfo", new EventConfig(`${BigNumber.from(this.depositAmount)} ${depositToken.symbol} has been deposited`, EventMessageType.Info, "Deposit completed"));
+    } else {
+      //something happened to make the deposit not happen
+      this.eventAggregator.publish("handleException", new EventConfigException("An error occurred while trying to deposit tokens. Please try again.", EventMessageType.Exception));
```
Comment (dkent600) [inline src/funding/funding.ts:229]: best to use "handleFailure" and just pass it a string for the message.

File: src/funding/funding.ts [hunk 2]
```diff
@@ -149,6 +147,12 @@ export class Funding {
    * @param transaction
    */
   public withdraw = async(transaction: IDaoTransaction): Promise<void> => {
```
Comment (dkent600) [inline src/funding/funding.ts:149]: can you define this function in our normal typescript way?

File: src/funding/funding.ts [hunk 3]
```diff
@@ -160,9 +164,18 @@ export class Funding {
...
-      //TODO wire up the withdraw method
-      this.eventAggregator.publish("handleException", new EventConfigException("This method is not implemented", EventMessageType.Exception));
+      //withdraw the tokens
+      const withdrawTransaction = await this.deal.withdrawTokens(transaction.dao, transaction.depositId);
+      if (withdrawTransaction){
+        //tokens were withdrawn successfully
+        this.eventAggregator.publish("handleInfo", new EventConfig("Your deposit has been withdrawn", EventMessageType.Info, "Withdrawn"));
+      } else {
+        //an error occurred while trying to withdraw tokens
+        this.eventAggregator.publish("handleException", new EventConfigException("An error occurred while trying to withdraw. Please try again.", EventMessageType.Exception));
```
Comment (dkent600) [inline src/funding/funding.ts:174]: Do "handleFailure" with just a string as the message.

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/funding/funding.ts (hunk 1 above)
Comment (vipergtsrz) [reply src/funding/funding.ts]: Done

File: src/funding/funding.ts (hunk 2 above)
Comment (vipergtsrz) [reply src/funding/funding.ts]: Done

File: src/funding/funding.ts (hunk 3 above)
Comment (vipergtsrz) [reply src/funding/funding.ts]: Done

**Pull request merged**